crates/aeronet_websocket/src/server/config.rs

```diff
@@ -111,15 +183,26 @@ impl ServerConfigBuilder<WantsTlsConfig> {
             bind_address: self.0.bind_address,
             tls,
             socket: WebSocketConfig::default(),
+            handshake_handler: None,
         }
     }
 }
 
 impl ServerConfig {
-    /// Configures this to use the given socket configuration.
+    /// Configures config to use the given socket configuration.
     pub fn with_socket_config(self, socket: WebSocketConfig) -> Self {
         Self { socket, ..self }
     }
+
+    /// Configures config to use the given handshake callback.
+    pub fn with_handshake_handler(self, handshake_handler: HandshakeHandler) -> ServerConfig {
+        ServerConfig {
+            bind_address: self.bind_address,
+            tls: self.tls,
+            socket: self.socket,
+            handshake_handler: Some(handshake_handler),
+        }
+    }
```

Comment:
use the same struct builder syntax as above
```suggestion
    /// Configures config to use the given handshake callback.
    pub fn with_handshake_handler(self, handshake_handler: HandshakeHandler) -> Self {
        Self {
            handshake_handler: Some(handshake_handler),
            ..self
        }
    }
```